controlplane/kubeadm/controllers/scale.go

```diff
@@ -149,5 +132,8 @@ func selectMachineForScaleDown(controlPlane *internal.ControlPlane) (*clusterv1.
 	if needingUpgrade := controlPlane.MachinesNeedingRollout(); needingUpgrade.Len() > 0 {
 		machines = needingUpgrade
 	}
+	if unhealthy := controlPlane.UnhealthyMachines(); unhealthy.Len() > 0 {
```

Comment:
Should unhealthy machines be prioritized here? It seems odd that we would potentially leave a machine failing MHC, but potentially remove a healthy machine in the case of an upgrade being in progress.

Reply:
I believe they are being prioritized, maybe I am misunderstanding the code or your question

Reply:
Ah indeed, my apologies, the overwriting of `machines` caused me a bit of confusion when I was going through the last review